### PR TITLE
Add some checks to guard against the project list becoming empty and …

### PR DESCRIPTION
**Pull request by nexus6-haiku**
…the project folder being deleted outside of Genio.

### CODE REVIEW

**Comment by nexus6-haiku** [comment]
Fixes #240 and #244

File: src/git/SourceControlPanel.cpp [hunk 1]
```diff
@@ -289,20 +291,21 @@ SourceControlPanel::MessageReceived(BMessage *message)
 					{
 						LogInfo("MSG_NOTIFY_PROJECT_SET_ACTIVE");
 						fSelectedProject = gMainWindow->GetActiveProject();
-						_UpdateProjectList();
+						if (!fProjectList->IsEmpty())
+							_UpdateBranchList();
```
Comment (jackburton79) [inline src/git/SourceControlPanel.cpp:295]: Is this change intended ? _UpdateProjectList() vs _UpdateBranchList()

Comment (nexus6-haiku) [reply src/git/SourceControlPanel.cpp:295]: It's an awful typo, sorry about that

**Comment by jackburton79** [comment]
Confirmed fixed #240 and #244
